src/Illuminate/Database/Console/Migrations/TableGuesser.php

```diff
@@ -10,8 +10,8 @@ class TableGuesser
     ];
 
     const CHANGE_PATTERNS = [
-        '/_(to|from|in)_(\w+)_table$/',
-        '/_(to|from|in)_(\w+)$/',
+        '/_(to|from|in)_(?!.*?(\_to\_|\_in\_|\_from\_))(\w+)_table$/',
+        '/_(to|from|in)_(?!.*?(\_to\_|\_in\_|\_from\_))(\w+)$/',
     ];
```

Comment:
I think we can just add `.*` in the beginning of the regex, so `'/_(to|from|in)_(\w+)$/'` becomes `'/.*_(to|from|in)_(\w+)$/'`. Based on my testing it also solves the issue and makes regex simpler.

Reply:
But better double check it before applying, maybe I missed something :)

Reply:
Overall thanks for PR, I think it's useful for cases like you described.

Reply:
@i350, good catch on the bug here. I agree with @hivokas though, this regex could be a little simpler, for example:

```php
    const CHANGE_PATTERNS = [
        '/.+_(to|from|in)_(\w+)_table$/',
        '/.+_(to|from|in)_(\w+)$/',
    ];
```

Given the _greedy_ nature of regular expressions, this will force it to only match the "last" instance in the string.